spyder/api/mode.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Copyright © <2013-2016> <Colin Duquesnoy and others, see pyqode/AUTHORS.rst>
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the editor extension API.
+Adapted from https://github.com/pyQode/pyqode.core/blob/master/pyqode/core/api/mode.py
+"""
+import logging
+
+
+def _logger():
+    return logging.getLogger(__name__)
+
+
+class Mode(object):
+    """
+    Base class for editor extensions. An extension is a "thing" that can be
+    installed on an editor to add new behaviours or to modify its appearance.
+
+    A mode is added to an editor by using the ModesManager/PanelsManager:
+
+        - :meth:`pyqode.core.api.CodeEdit.modes.append` or
+        - :meth:`pyqode.core.api.CodeEdit.panels.append`
+
+    Subclasses may/should override the following methods:
+
+        - :meth:`pyqode.core.api.Mode.on_install`
+        - :meth:`pyqode.core.api.Mode.on_uninstall`
+        - :meth:`pyqode.core.api.Mode.on_state_changed`
+
+    ..warning: The mode will be identified by its class name, this means that
+    **there cannot be two modes of the same type on the same editor instance!**
```

Comment:
Please rewrite this docstring too to refer to Spyder objects instead of PyQode ones